Test failure on Windows - TestDockerCLICheckCacheFromArgs

Currently `TestDockerCLICheckCacheFromArgs` is failing on Windows, sample failure output below:
```
--- FAIL: TestDockerCLICheckCacheFromArgs (0.04s)
13    --- FAIL: TestDockerCLICheckCacheFromArgs/multiple_cache-from_images (0.01s)
14        cmd_helper.go:208: 
15            wanted: docker build . --file C:\Users\RUNNER~1\AppData\Local\Temp\skaffold2147190149\Dockerfile -t tag --cache-from from/image1 --cache-from from/image2
16
17               got: docker build . --file C:\Users\RUNNER~1\AppData\Local\Temp\skaffold2147190149\C:\Users\RUNNER~1\AppData\Local\Temp\skaffold2147190149\Dockerfile -t tag --cache-from from/image1 --cache-from from/image2
18    --- FAIL: TestDockerCLICheckCacheFromArgs/cache-from_self_uses_tagged_image (0.02s)
19        cmd_helper.go:208: 
20            wanted: docker build . --file C:\Users\RUNNER~1\AppData\Local\Temp\skaffold2487207556\Dockerfile -t gcr.io/k8s-skaffold/test:tagged --cache-from gcr.io/k8s-skaffold/test:tagged
21
22               got: docker build . --file C:\Users\RUNNER~1\AppData\Local\Temp\skaffold2487207556\C:\Users\RUNNER~1\AppData\Local\Temp\skaffold2487207556\Dockerfile -t gcr.io/k8s-skaffold/test:tagged --cache-from gcr.io/k8s-skaffold/test:tagged

```

@marlon-gamez did some digging here and found the likely root cause:

> but after poking around a bit this kind of seems like  an expected failure, since we use filepath.Abs(dockerfilePath), which simply checks for a "/" prefix (even on windows). Since we're passing an absolute windows path, filepath.Abs() concats the working dir with the absolute windows path we gave

--- a/pkg/skaffold/docker/dependencies.go
+++ b/pkg/skaffold/docker/dependencies.go
@@ -21,10 +21,12 @@ import (
 	"fmt"
 	"os"
 	"path/filepath"
+	"runtime"
 	"sort"
 
 	"github.com/docker/docker/builder/dockerignore"
 
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/constants"
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/walk"
 )
@@ -61,6 +63,9 @@ func NormalizeDockerfilePath(context, dockerfile string) (string, error) {
 		if _, err := os.Stat(dockerfile); err == nil || !os.IsNotExist(err) {
 			return filepath.Abs(dockerfile)
 		}
+	}
+	if runtime.GOOS == constants.Windows && (filepath.VolumeName(dockerfile) != "" || filepath.IsAbs(dockerfile)) {
+		return dockerfile, nil
 	}
 	return filepath.Abs(rel)
 }

